server/src/services/restaurant.ts

```diff
@@ -116,19 +121,59 @@ function generateNearbyQuery(
   ];
 }
 
+// Core of pagination
+// Returns the total amount of restaurants that match the filter
+// Limits the amount of restaurants retrieved
+function generateRestaurantLimitQuery(
+  skippedDocuments: number,
+  documentLimit: number
+) {
+  const limitQuery = [
+    {
+      $facet: {
+        restaurants: [{ $skip: skippedDocuments }, { $limit: documentLimit }],
+        count: [{ $group: { _id: null, count: { $sum: 1 } } }],
+      },
+    },
+    {
+      $unwind: "$count",
+    },
+    {
+      $project: {
+        restaurants: true,
+        count: "$count.count",
+      },
+    },
+  ];
+
+  return limitQuery;
+}
+
 async function findNear(
   coordinates: Coordinates,
-  searchRadius: Meters
-): Promise<RestaurantResult[]> {
+  searchRadius: Meters,
+  skip: number,
+  limit: number
+): Promise<NearbyRestaurantsResult> {
+  console.log({ skip, limit });
```

Comment:
any reason to have this? or just needed for dev purposes?

Reply:
good catch, forgot to remove that